src/main/java/com/alphawallet/attestation/IdentifierAttestation.java

```diff
@@ -61,6 +61,24 @@ public IdentifierAttestation(byte[] commitment, AsymmetricKeyParameter key)  {
     setCommitment(commitment);
   }
 
+  public IdentifierAttestation(String type, String identifier, AsymmetricKeyParameter key)  {
+    super();
+    super.setVersion(18); // Our initial version
+    super.setSubject("CN=" + SignatureUtility.addressFromKey(key));
+    super.setSigningAlgorithm(SignatureUtility.ALGORITHM_IDENTIFIER);
+    try {
+      SubjectPublicKeyInfo spki = SubjectPublicKeyInfoFactory.createSubjectPublicKeyInfo(key);
+      super.setSubjectPublicKeyInfo(spki);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    //Add the identifier in the form 'TW=@twittertest' or 'EMAIL=test@test.org'
+    ASN1EncodableVector dataObject = new ASN1EncodableVector();
+    dataObject.add(new DEROctetString((type + "=" + identifier).getBytes()));
+    super.setDataObject(new DERSequence(dataObject));
```

Comment:
This should work just fine, but I think maybe it is better to also store this as an extension in a similar manner to the commitment, because this allows us to follow the x509 standard more closely. But for all our internal code this should work just fine